01_fib/patrickmarabeas/main.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+// Fibonacci slice generator
+// - Starts from 0
+// - Handles both positive and negative numbers
+// - Returns nil for input of 0
+// - Returns an error if the input would generate numbers outside of an int lower/upper values
+func calculateFibSequence(n int) ([]int, error) {
+	// Multiplied against the initial Fibonacci value (1) to spawn either a positive or negative sequence
+	sign := 1
+	// Early bailout returning zero value of a slice
+	if n == 0 {
+		return nil, nil
+	}
+	// Early bailout and error for return values outside of int limits
+	if n > 92 || n < -92 {
+		return nil,
+			fmt.Errorf(`ERROR: %d is outside of limits Please enter a number between -92 and 92`, n*sign)
+	}
+	// Handle negative inputs
+	if n < 0 {
+		n *= -1
+		sign = -1
+	}
+	// Return a single length slice of zero value int
+	if n <= 1 {
+		return make([]int, 1), nil
+	}
+	// Make a splice of required length
+	result := make([]int, n)
+	// Seed initial two values of the Fibonacci sequence; accommodating for the negative sequence
+	result[0] = 0
+	result[1] = 1 * sign
+	// Calculate remaining sequence
+	for i := 2; i < n; i++ {
+		result[i] = result[i-2] + result[i-1]
+	}
+
+	return result, nil
+}
+
+// Fibonacci IO
+func fib(n int) {
+	seq, err := calculateFibSequence(n)
+	if err != nil {
+		fmt.Fprintln(out, err)
+	} else {
+		for _, num := range seq {
+			fmt.Fprintln(out, num)
+		}
+	}
+}
+
+func main() {
+	fib(7)
+	fib(100)
```

Comment:
Technically the lab asks for the output of `fib(7)` only. I don't care all that much though, but if you are going to deviate from a spec, best to add a comment as to why so we know it's deliberate and not an accidental deviation.

Reply:
Resolved; added test case instead.